lib/core/src/Cardano/Wallet/Registry.hs

```diff
@@ -195,10 +193,9 @@ defaultWorkerAfter
     -> IO ()
 defaultWorkerAfter tr = traceWith tr . \case
     Right _ -> MsgFinished
-    Left e -> case asyncExceptionFromException e of
-        Just ThreadKilled -> MsgThreadKilled
-        Just UserInterrupt -> MsgUserInterrupt
-        _ -> MsgUnhandledException $ pretty $ show e
+    Left e -> if isSyncException e
+        then MsgThreadCancelled
+        else MsgUnhandledException $ pretty $ show e
```

Comment:
:+1: ok

Reply:
Ah, no. Not okay. The `then` and `else` clauses are mixed up.

Reply:
Fixed in: https://github.com/input-output-hk/cardano-wallet/compare/fe58426a3cd9609800c4eac9b5662f74bdb2a4dd..db3dc33736b7dd6424bb2afa32a88def19bacd20